core/src/test/java/io/questdb/griffin/engine/functions/str/SplitPartCharFunctionFactoryTest.java

```diff
@@ -0,0 +1,97 @@
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.str;
+
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.engine.AbstractFunctionFactoryTest;
+import io.questdb.griffin.engine.functions.constants.CharConstant;
+import io.questdb.std.Numbers;
+import io.questdb.test.tools.TestUtils;
+import org.junit.Test;
+
+public class SplitPartCharFunctionFactoryTest extends AbstractFunctionFactoryTest {
+
+    @Override
+    protected FunctionFactory getFunctionFactory() {
+        return new SplitPartCharFunctionFactory();
+    }
+
+    @Test
+    public void testPositiveIndex() throws SqlException {
+        assertQuery(
+                "split_part\n" + "def\n",
+                "select split_part('abc,def,ghi,jkl', ',', 2)",
+                null,
+                true,
+                true);
+    }
+
+    @Test
+    public void testNegativeIndex() throws SqlException {
+        assertQuery(
+                "split_part\n" + "ghi\n",
+                "select split_part('abc,def,ghi,jkl', ',', -2)",
+                null,
+                true,
+                true);
+    }
+
+    @Test
+    public void testNullOrEmptyStr() throws SqlException {
+        callCustomised(true, true, null, ',', 2).andAssert(null);
+        callCustomised(true, true, "", ',', 2).andAssert("");
+    }
+
+    @Test
+    public void testZeroDelimiter() throws SqlException {
+        callCustomised(true, true, "abc,def,ghi,jkl", CharConstant.ZERO.getChar(null), 2).andAssert(null);
+    }
+
+    @Test
+    public void testNaNIndex() throws SqlException {
+        callCustomised(true, true, "abc,def,ghi,jkl", ',', Numbers.INT_NaN).andAssert(null);
+    }
+
+    @Test(expected = SqlException.class)
+    public void testDynamicIndex() throws SqlException {
+        try {
+            call("abc,def,ghi,jkl", ',', 2);
+        } catch (SqlException e) {
+            TestUtils.assertContains(e.getFlyweightMessage(), "index must be a constant or runtime-constant");
+            throw e;
```

Comment:
nit: re-throwing could be replaced with an `Assert.fail()` in the try block.

Reply:
The rethrow is required because the test expects `SqlException` to be thrown. Notice the test with annotated with `@Test(SqlException.class)`. But we need to assert the error message so that a different `SqlException` thrown does not cause a false positive.

Reply:
Yes, I've noticed that. Our usual approach is have a `Assert.fail()` and no `expected` value.

Reply:
Ah I see what you mean now, I'll change it